src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -349,6 +354,14 @@ protected virtual void Initialize()
                     TraceWriter = new ConsoleTraceWriter(hostTraceLevel);
                 }
 
+                // Use the startupLogger in this class as it is concerned with startup. The public Logger is used
+                // for all other logging after startup. There's no filters at this point, but that's okay since we
+                // can't build filters until we parse host.json below.
+                // Initialize the public Logger to the startup logger until we've read configuration settings and
+                // can create the real logger.
+                ConfigureDefaultLoggerFactory();
```

Comment:
@brettsam Can you take a look at this refactoring? I did this so we had a default logger in place for error paths.

Reply:
This should work, but you'll need to re-create the `_hostConfig.LoggerFactory` and call `ConfigureDefaultLoggerFactory` again after host.json is applied. That will re-build the factory with the settings that we read (applying filters and AppInsights values, etc).

If you move the newing-up of that LoggerFactory into `ConfigureDefaultLoggerFactory` itself, it should be something like:
```
ApplyConfiguration(hostConfigObject, ScriptConfig);

// this lets tests change anything parsed from host.json so they don't have to overwrite the file themselves
ScriptConfig.OnConfigurationApplied?.Invoke(ScriptConfig); 

// now the scriptconfig should be final, so re-create the loggerfactory and loggers
CreateDefaultLoggerFactory();

_startupLogger = _hostConfig.LoggerFactory.CreateLogger(LogCategories.Startup);
Logger = _hostConfig.LoggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
HostInitializing?.Invoke(this, EventArgs.Empty);
```

Reply:
Done